### ISSUE TITLE
Rule proposal: prefer-split-limit

### ISSUE THREAD

**Issue by gurgunday** [sindresorhus/eslint-plugin-unicorn#2464]
### Description

Whenever we use `String.prototype.split` with literals like so:

```js
string.split('/')[0]
// or
string.split('/')[1]
// or
string.split('/')[2]
```

We benefit by limiting the search space, this allows split to exit early once it attains the number of splits and to not scan the whole string:

```js
string.split('/', 1)[0]
// and
string.split('/', 2)[1]
// and
string.split('/', 3)[2]

// Once we get to the element we want, .split exits early (faster)
```

### Fail

```js
string.split('/')[0]
```

### Pass

```js
string.split('/', 1)[0]
```

### Proposed rule name

specify-split-limit or prefer-split-limit

**Comment by fregante** [comment]
> ```js
> string.split('/').at(numberVar)
> ```

Could also be:

```js
string.split('/', numberVar).at(numberVar)
```

**Comment by gurgunday** [comment]
Yeah

Just a small correction in case anyone starts implementing the rule -- since the second parameter is the length of the returned array instead of the number of splits to do:

```js
string.split('/').at(numberVar)

// would be:

string.split('/', numberVar + 1).at(numberVar)
```

**Comment by fregante** [comment]
The rule could potentially also warn for that:

```js
s.split('/', 1).at(2);
// Error: This will always be undefined

**Comment by sindresorhus** [comment]
Accepted

**Comment by fisker** [comment]
> > string.split('/').at(numberVar)
> 
> Could also be:
> 
> string.split('/', numberVar).at(numberVar)



> > string.split('/').at(numberVar)
> 
> Could also be:
> 
> string.split('/', numberVar).at(numberVar)

Don't think it's safe, since `numberVar` can be negative numbers.
